File: ios/Sasquatsh/Views/Groups/RecurringGameCard.swift
import SwiftUI

struct RecurringGameCard: View {
    let game: RecurringGame
    let isAdmin: Bool
    var onEdit: (() -> Void)?
    var onToggleActive: (() -> Void)?
    var onDelete: (() -> Void)?

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack {
                Text(game.title)
                    .font(.md3TitleMedium)
                    .foregroundStyle(Color.md3OnSurface)
                Spacer()
                BadgeView(
                    text: game.isActive ? "Active" : "Paused",
                    color: game.isActive ? .md3PrimaryContainer : .md3SecondaryContainer
                )
            }

            Label(game.scheduleDescription, systemImage: "calendar.badge.clock")
                .font(.md3BodyMedium)
                .foregroundStyle(Color.md3OnSurfaceVariant)

            HStack(spacing: 8) {
                if let system = game.gameSystem, !system.isEmpty,
                   let gs = GameSystem(rawValue: system), gs != .boardGame {
                    BadgeView(text: gs.shortName, color: gs.badgeColor)
                }

                if let gameTitle = game.gameTitle, !gameTitle.isEmpty {
                    BadgeView(text: gameTitle, color: .md3SurfaceContainerHigh)
                }

                BadgeView(text: "\(game.durationMinutes) min", color: .md3SurfaceContainerHigh)
                BadgeView(text: "\(game.maxPlayers) players", color: .md3SurfaceContainerHigh)
            }

            if let city = game.city, let state = game.state {
                Label("\(city), \(state)", systemImage: "mappin")
                    .font(.md3BodySmall)
                    .foregroundStyle(Color.md3OnSurfaceVariant)
            } else if let details = game.locationDetails, !details.isEmpty {
                Label(details, systemImage: "mappin")
                    .font(.md3BodySmall)
                    .foregroundStyle(Color.md3OnSurfaceVariant)
            }

            if let nextDate = game.nextOccurrenceDate {
                Label("Next: \(nextDate.toDate?.displayDate ?? nextDate)", systemImage: "arrow.right.circle")
                    .font(.md3BodySmall)
                    .foregroundStyle(Color.md3Primary)
            }
        }
        .padding()
        .cardStyle()
        .contextMenu {
            if isAdmin {
                Button { onEdit?() } label: {
                    Label("Edit", systemImage: "pencil")
                }
                Button {
                    onToggleActive?()
                } label: {
                    Label(game.isActive ? "Pause" : "Resume", systemImage: game.isActive ? "pause.circle" : "play.circle")
                }
                Button(role: .destructive) { onDelete?() } label: {
                    Label("Delete", systemImage: "trash")
                }
            }
        }
    }
}

File: android/app/src/main/java/com/sasquatsh/app/views/groups/RecurringGameCard.kt
package com.sasquatsh.app.views.groups

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowForward
import androidx.compose.material.icons.filled.Delete
import androidx.compose.material.icons.filled.Edit
import androidx.compose.material.icons.filled.LocationOn
import androidx.compose.material.icons.filled.PauseCircle
import androidx.compose.material.icons.filled.PlayCircle
import androidx.compose.material.icons.filled.Schedule
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.DropdownMenu
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.sasquatsh.app.models.GameSystem
import com.sasquatsh.app.models.RecurringGame
import com.sasquatsh.app.views.events.formatEventDate
import com.sasquatsh.app.views.shared.BadgeView

@Composable
fun RecurringGameCard(
    game: RecurringGame,
    isAdmin: Boolean,
    modifier: Modifier = Modifier,
    onEdit: (() -> Unit)? = null,
    onToggleActive: (() -> Unit)? = null,
    onDelete: (() -> Unit)? = null
) {
    var showMenu by remember { mutableStateOf(false) }

    Card(
        onClick = { if (isAdmin) showMenu = true },
        modifier = modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceContainer),
        shape = RoundedCornerShape(12.dp),
        elevation = CardDefaults.cardElevation(defaultElevation = 1.dp)
    ) {
        Column(
            modifier = Modifier.padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            // Title + status badge
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = game.title,
                    style = MaterialTheme.typography.titleMedium,
                    color = MaterialTheme.colorScheme.onSurface,
                    modifier = Modifier.weight(1f)
                )
                BadgeView(
                    text = if (game.isActive) "Active" else "Paused",
                    color = if (game.isActive) {
                        MaterialTheme.colorScheme.primaryContainer
                    } else {
                        MaterialTheme.colorScheme.secondaryContainer
                    }
                )
            }

            // Schedule
            Row(
                horizontalArrangement = Arrangement.spacedBy(6.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Icon(
                    imageVector = Icons.Filled.Schedule,
                    contentDescription = null,
                    tint = MaterialTheme.colorScheme.onSurfaceVariant,
                    modifier = Modifier.size(16.dp)
                )
                Text(
                    text = game.scheduleDescription,
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }

            // Badges row
            Row(
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                // Game system
                val gs = game.gameSystem?.let { GameSystem.fromValue(it) }
                if (gs != null && gs != GameSystem.BOARD_GAME) {
                    BadgeView(text = gs.shortName)
                }

                // Game title
                val gameTitle = game.gameTitle
                if (!gameTitle.isNullOrEmpty()) {
                    BadgeView(text = gameTitle)
                }

                // Duration
                BadgeView(text = "${game.durationMinutes} min")

                // Max players
                BadgeView(text = "${game.maxPlayers} players")
            }

            // Location
            val city = game.city
            val state = game.state
            val locationDetails = game.locationDetails
            if (city != null && state != null) {
                Row(
                    horizontalArrangement = Arrangement.spacedBy(4.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        Icons.Filled.LocationOn, null,
                        tint = MaterialTheme.colorScheme.onSurfaceVariant,
                        modifier = Modifier.size(14.dp)
                    )
                    Text(
                        text = "$city, $state",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            } else if (!locationDetails.isNullOrEmpty()) {
                Row(
                    horizontalArrangement = Arrangement.spacedBy(4.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        Icons.Filled.LocationOn, null,
                        tint = MaterialTheme.colorScheme.onSurfaceVariant,
                        modifier = Modifier.size(14.dp)
                    )
                    Text(
                        text = locationDetails,
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            }

            // Next occurrence
            val nextDate = game.nextOccurrenceDate
            if (nextDate != null) {
                Row(
                    horizontalArrangement = Arrangement.spacedBy(4.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        Icons.Filled.ArrowForward, null,
                        tint = MaterialTheme.colorScheme.primary,
                        modifier = Modifier.size(14.dp)
                    )
                    Text(
                        text = "Next: ${formatEventDate(nextDate)}",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.primary
                    )
                }
            }
        }

        // Context menu for admins
        if (isAdmin) {
            DropdownMenu(expanded = showMenu, onDismissRequest = { showMenu = false }) {
                DropdownMenuItem(
                    text = { Text("Edit") },
                    leadingIcon = { Icon(Icons.Filled.Edit, null) },
                    onClick = {
                        showMenu = false
                        onEdit?.invoke()
                    }
                )
                DropdownMenuItem(
                    text = { Text(if (game.isActive) "Pause" else "Resume") },
                    leadingIcon = {
                        Icon(
                            if (game.isActive) Icons.Filled.PauseCircle else Icons.Filled.PlayCircle,
                            null
                        )
                    },
                    onClick = {
                        showMenu = false
                        onToggleActive?.invoke()
                    }
                )
                DropdownMenuItem(
                    text = { Text("Delete", color = MaterialTheme.colorScheme.error) },
                    leadingIcon = {
                        Icon(
                            Icons.Filled.Delete, null,
                            tint = MaterialTheme.colorScheme.error
                        )
                    },
                    onClick = {
                        showMenu = false
                        onDelete?.invoke()
                    }
                )
            }
        }
    }
}
